astronomy-pipeline/astronomy_transform_functions.py

```diff
@@ -0,0 +1,85 @@
+"""Functions for the astronomy API transform script."""
+
+import json
+import pandas as pd
+
+from astronomy_extract_functions import get_db_connection
+
+
+def load_from_file(filename: str) -> dict:
+    """Load the stories from a file called stories.json"""
+
+    with open(filename, "r", encoding="UTF-8") as f_obj:
+        return json.load(f_obj)
+
+
+def get_data_into_database(raw_data: dict) -> pd.DataFrame:
+    """Converts a list of body position dictionaries into a dataframe."""
+
+    regions = list(raw_data["body_positions"].keys())
+    times = list(raw_data["body_positions"]["1"].keys())
+
+    df_list = []
+    for r in regions:
+        for t in times:
+            time_list = raw_data["body_positions"][r][t]
+            df = pd.DataFrame(time_list)
+            df["region_id"] = int(r)
+            df_list.append(df)
+
+    return pd.concat(df_list, ignore_index=True)
+
+
+def get_body_mapping() -> dict:
+    """Returns body name to ID mapping from the RDS database."""
+
+    with get_db_connection() as conn:
+
+        cur = conn.cursor()
+
+        cur.execute("SELECT body_id, body_name from body;")
+
+        results = cur.fetchall()
```

Comment:
Does the cursor need to be explicitly closed here or it it okay because it's attached to the connection which is closed?

Reply:
Yes I think it closes automatically.

Reply:
Good point by Caleb, would be better for the cursor to be either explicitly closed or indented into another with block.         

```
# Open the database connection
        with get_db_connection() as conn:
            # Open and close the cursor using a with statement
            with conn.cursor() as cur: 
```

It should still be okay as the cursor in the context of the connection .